IBM/ACE-RISCV
# Symmetrical multiprocessing

## Description of the changes
Enable symmetrical multiprocessing in confidential VMs.

## Type of change

## How to test this PR?
Run the system in emulator using `ace run` command. Observe that during the boot process, two harts are booting up and execute the ACE initialization procedure
```
#ACE: Setting up hardware hart id=0
#ACE: Hardware hart id=0 has state area region at 1800101c0
#ACE: Hardware hart id=0 registered trap handler at address: 80019280
#ACE: Setting up hardware hart id=1
#ACE: Hardware hart id=1 has state area region at 180010988
#ACE: Hardware hart id=1 registered trap handler at address: 80019280
```

The hypervisor should also report that it has 2 CPUs
```
[    0.061435] smp: Bringing up secondary CPUs ...
[    0.071932] cacheinfo: Unable to detect cache hierarchy for CPU 1
[    0.081303] smp: Brought up 1 node, 2 CPUs
```

After login to the hypervisor, start the test VM using the command `./run_linux_vm.sh`. Check that the VM starts two CPUs
```
Number of cores assigned to the guest: 2

[    6.389983] smp: Bringing up secondary CPUs ...
[    6.400707] cacheinfo: Unable to detect cache hierarchy for CPU 1
[    6.424520] smp: Brought up 1 node, 2 CPUs
```
